asf_tools/hand/calculate.py

```diff
@@ -30,14 +30,31 @@ def fill_nan(array: np.ndarray) -> np.ndarray:
     kernel = astropy.convolution.Gaussian2DKernel(x_stddev=3)  # kernel x_size=8*stddev
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-
-        array = astropy.convolution.interpolate_replace_nans(
-            array, kernel, convolve=astropy.convolution.convolve
-        )
+        while np.any(np.isnan(array)):
```

Comment:
Instead of inside the filling function, I'd prefer to see this `while` loop wrapped around the call to `fill_hand` on line 127:
https://github.com/ASFHyP3/asf-tools/blob/217e4d08020edb7dec8e48b00d39df4b806c8566/asf_tools/hand/calculate.py#L127

That makes this function easier to test, more straightforward to use, and makes it easier to control the number of iterations

Reply:
Never mind; looking at the code more closely I think this is better.